src/stores/asynchronousAction.ts

```diff
@@ -10,13 +14,21 @@ export interface IAsynchronousAction<S, R> {
   // and it is not a real "action" that a reducer will ever see.
   type: string;
   types: IActionTypeCollection;
-  doPerform: (state: S) => Promise<R> | undefined;
+  doPerform: (
+    state: S,
+    dispatch: ThunkDispatch<IState, void, AnyAction>
+  ) => Promise<R> | undefined;
```

Comment:
Does this make sense with the way you designed it?
```suggestion
  ) => Promise<R> | void;
```